doc/docusaurus/docs/reference/CIP-0069-changes.md

```diff
@@ -0,0 +1,177 @@
+---
+sidebar_position: 27
+---
+
+# Understanding CIP-0069 changes
+
+## Overview
+
+Cardano Improvement Proposal 69 [(CIP-0069)](https://cips.cardano.org/cip/CIP-0069) introduces significant changes to how Plutus scripts are written and executed. 
+It streamlines the development and functionality of Plutus scripts by unifying the arguments that are provided to different types of scripts. 
+Specifically, it removes the datum argument, which has been a source of complexity and limitations in decentralized application (DApp) development on the Cardano blockchain.
+These changes also reduce the chances of attacks succeeding. 
+
+We will review these changes and take a look at how to implement them.
+
+## 1. Unified signature for all Plutus scripts
+
+### What changed
+
+Previously, different types of scripts (like spending or minting) had different structures, making it challenging to create multi-purpose scripts. 
+CIP-0069 introduces a unified signature for all Plutus scripts, regardless of their purpose.
+This unification simplifies the Plutus codebase and allows for more flexible script design.
+
+### Implementation of new signature
+To implement this change, it is necessary to update all scripts to use this new signature:
+```haskell
+ScriptArgs -> ScriptContext -> ()
+```
+
+### Example
+
+Here's an example of a unified script: 
+
+```haskell
+{-# INLINABLE unifiedScript #-}
+unifiedScript :: ScriptArgs -> ScriptContext -> ()
+unifiedScript args ctx = 
+  case args of
+    RedeemerOnly r -> handleRedeemer r ctx
+    RedeemerAndDatum r d -> handleBoth r d ctx
+  where
+    handleRedeemer r ctx = ...
+    handleBoth r d ctx = ...
+```
+
+This code shows how a script now handles different types of inputs (`RedeemerOnly` or `RedeemerAndDatum`) within a single function.
+
+### Key points
+
+- The `ScriptArgs` type allows for handling both Redeemer-only and Redeemer-with-Datum cases.
+- `ScriptContext` now contains all the contextual information, including what was previously passed as separate arguments.
+- Scripts must return `()` (unit) to indicate successful validation.
```

Comment:
This could have been true if only CIP-117 wasn't implemented, which mandates return type to be `BuiltinUnit`

Reply:
Alright, I'll correct this, good catch.